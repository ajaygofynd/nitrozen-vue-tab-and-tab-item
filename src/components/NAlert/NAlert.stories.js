/* Component imports */
import NitrozenAlert from './NAlert.vue';

export default {
    title: 'Components/Alert',
    component: NitrozenAlert,
    argTypes: {
        alertWidth: {
            control: 'text',
            description: `This property sets the width of an alert.`
        },
        buttonText: {
            control: 'text',
            description: `This property sets the text of the button, if the button is displayed.`
        },
        buttonType: {
            control: 'select',
            description: `This property sets the shape of the button, as a linked text or a rectangle box.`,
            options: ['button', 'default', 'link']
        },
        displayButton: {
            control: 'boolean',
            description: `This property decides whether the button should be displayed or hidden.`
        },
        extendedAlert: {
            control: 'boolean',
            description: `This property decides whether the alert should be shown in an extended format, like a card.`
        },
        fullWidth: {
            control: 'boolean',
            defaultValue: true,
            description: `This property decides whether the component should take up the entire width of its parent component.`
        },
        href: {
            control: 'text',
            description: `This property will be the URL that needs to be navigated to on the click of a link button, in scenarios where the button type is selected as a link.`
        },
        labelText: {
            control: 'text',
            description: `This property will set the value of the label in the alert.`
        },
        linkText: {
            control: 'text',
            description: `This property will set the value of the link button for scenarios where the button type is a link.`
        },
        loader: {
            control: 'boolean',
            description: `This property decides if the alert needs to be displayed as a loader.`
        },
        state: {
            name: 'state',
            control: 'select',
            defaultValue: 'info',
            description: `This property decides the state of the alert.`,
            options: ['error', 'info', 'success', 'warn'],
            type: {
                name: 'string',
                required: true
            }
        }
    }
};

const Template = (args, {argTypes}) => ({
    components: { NitrozenAlert },
    props: Object.keys(argTypes),
    template: `
        <div 
            class="main-div space-between"
            style="display:flex; justify-content:space-evenly; flex-wrap:wrap; gap:0.4em;">
            <nitrozen-alert v-bind="$props" classNames="my-alert">
                This is a stateless (default) alert. Any change in the controls below will change this alert.
            </nitrozen-alert> 
            <nitrozen-alert v-bind="$props" state="error">
                This is an error alert.
            </nitrozen-alert> 
            <nitrozen-alert v-bind="$props" state="info">
                This is an informational alert.
            </nitrozen-alert> 
            <nitrozen-alert v-bind="$props" state="success">
                This is a success alert.
            </nitrozen-alert> 
            <nitrozen-alert v-bind="$props" state="warn">
                This is a warning alert.
            </nitrozen-alert> 
        </div>
    `
});

/**
 * An alert without a button
 */
export const ButtonLessAlert = Template.bind({});
ButtonLessAlert.args = {
    displayButton: false
};
ButtonLessAlert.storyName = 'Alert without button';

/**
 * An alert with a button
 */
export const ButtonAlert = Template.bind({});
ButtonAlert.args = {
    buttonText: "Accept",
    buttonType: "default",
    displayButton: true
};
ButtonAlert.storyName = "Alert with button";

/**
 * An alert with a link button
 */
export const ButtonLink = Template.bind({});
ButtonLink.args = {
    buttonType: "link",
    displayButton: true,
    href: "https://www.google.com",
    linkText: "Next"
};
ButtonLink.storyName = "Alert with link";

/**
 * An alert with full width and without a button
 */
export const AlertFullWidth = Template.bind({});
AlertFullWidth.args = {
    displayButton: false,
    fullWidth: true
};
AlertFullWidth.storyName = "Alert without button and with full width";

/**
 * An alert without a button and without a full width (width set to auto)
 */
export const AlertAutoWidth = Template.bind({});
AlertAutoWidth.args = {
    displayButton: false,
    fullWidth: false
};
AlertAutoWidth.storyName = "Alert without button and without full width";

/**
 * An alert with a set width
 */
export const AlertSetWidth = Template.bind({});
AlertSetWidth.args = {
    alertWidth: "45%",
    displayButton: false,
    fullWidth: false
};
AlertSetWidth.storyName = "Alert without button and with a set width";

/**
 * An alert with a button and auto width
 */
export const ButtonAlertAutoWidth = Template.bind({});
ButtonAlertAutoWidth.args = {
    buttonType: "button",
    buttonText: "Click",
    displayButton: true,
    fullWidth: false
};
ButtonAlertAutoWidth.storyName = "Alert with button and without full width";

/**
 * An extended alert without a button
 */
export const ButtonLessExtendedAlert = Template.bind({});
ButtonLessExtendedAlert.args = {
    extendedAlert: true
};
ButtonLessExtendedAlert.storyName = "Extended Alert without button";

/**
 * An extended alert with a button
 */
export const ButtonExtendedAlert = Template.bind({});
ButtonExtendedAlert.args = {
    buttonText: 'Click',
    displayButton: true,
    extendedAlert: true
};
ButtonExtendedAlert.storyName = "Extended Alert with button";

/**
 * An alert with a loader
 */
export const LoaderAlert = Template.bind({});
LoaderAlert.args = {
    loader: true
};
LoaderAlert.storyName = "Alert with loader";

/**
 * An alert with custom class-name
 */
export const AlertCustomClass = Template.bind({});
AlertCustomClass.args = {
    className: "nitrozen-custom-class",
};
AlertCustomClass.storyName = "Alert with a custom classname";
