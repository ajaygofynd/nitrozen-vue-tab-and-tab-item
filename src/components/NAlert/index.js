import NAlert from './NAlert.vue';

export default NAlert;