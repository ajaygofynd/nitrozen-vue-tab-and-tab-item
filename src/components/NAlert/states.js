/**
 * This file is for storing state variables that will be 
 * used throughout the alert component.
 * This is to ensure that there are no spelling mistakes
 * or discrepencies while coding the component.
 * Furthermore, any new state or modification to the existing
 * state will be handled in one single place.
 * 
 * @author Rushabh Mulraj Shah
 * @since 0.0.42
 */

export const ERROR = 'error';
export const INFO = 'info';
export const SUCCESS = 'success';
export const WARN = 'warn';
