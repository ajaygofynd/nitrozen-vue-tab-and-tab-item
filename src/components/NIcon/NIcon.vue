<template>
    <svg :width="size" :height="size" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg" preserveAspectRatio="none">
        <path :id="name" :d="getSVG" :fill="color" />
    </svg>
</template>

<script>
import icons from '../../assets/icons.json'
export default {
    name: 'nitrozen-icon',
    props: {
        name: {
            type: String,
            required: true,
            default: 'info'
        },
        size: {
            type: Number,
            required: false,
            default: 32
        },
        color: {
            type: String,
            required: false,
            default: '#000000'
        }
    },
    computed: {
        getSVG() {
            return icons[this.$props.name];
        }
    }
}
</script>