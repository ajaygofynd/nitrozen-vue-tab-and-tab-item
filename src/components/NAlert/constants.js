/**
 * This file will be a centralised place for storing
 * all variables that will be used throughout the
 * Alert component.
 * 
 * @author Rushabh Mulraj Shah
 * @since 0.0.42
 */

/* Variable imports */
import * as STATES from './states';

export const ICON_NAMES = {
    [STATES.ERROR]: 'error_colored',
    [STATES.INFO]: 'info',
    [STATES.SUCCESS]: 'success_colored',
    [STATES.WARN]: 'warning_colored'
};

export const ICON_COLORS = {
    [STATES.ERROR]: '#F50031',
    [STATES.INFO]: '#3535F3',
    [STATES.SUCCESS]: '#219653',
    [STATES.WARN]: '#F06D0F'
};
