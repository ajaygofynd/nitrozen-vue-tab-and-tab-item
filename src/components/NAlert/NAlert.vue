<template>
    <transition name="nitrozen-alert">
        <div 
            class="n-alert"
            :class="[
                classes.container, 
                classNames,
                fullWidth && 'n-alert-full-width',
                buttonType === 'link' && 'n-alert-link-button-container',
                loader && 'n-alert-loader-container',
                alertWidth && 'n-alert-custom-width',
                isAlertXS && 'n-alert-xs'
            ]"
            :style="{
                width: componentWidth,
                height: extendedAlert ? 'auto' : 'fit-content'
            }">
            <div class="n-alert-content" :class="extendedAlert && 'n-alert-extended'">
                <div class="n-alert-icon-text-wrapper">
                    <div class="n-alert-icon">
                        <img :src="getLoader" v-if="loader" class="n-alert-loader" />
                        <nitrozen-icon 
                            v-else
                            :color="iconColor" 
                            :name="alertIcon" 
                            :size="26.67" />
                    </div>
                    <div className="n-alert-label-text" v-if="labelText">{{labelText}}</div>
                    <slot></slot>
                </div>

                <div class="n-alert-button-container" v-if="displayButton && !loader">
                    <button 
                        v-if="buttonType === 'default' || buttonType === 'button'"
                        class="n-alert-button"
                        :class="[classes.button]"
                        @click="onButtonClick"
                    >
                        {{ buttonText }}
                    </button>
                    <a
                        v-else-if="buttonType === 'link'"
                        class="n-alert-link"
                        :class="[classes.buttonLabel]"
                        @click="onButtonClick"
                        :href="href"
                    >
                        {{ linkText || buttonText || "Click" }}
                    </a>
                </div>
            </div>
        </div>
    </transition>
</template>

<script>
/* Variables and constants imports */
import * as STATES from './states';
import { LOADER_CDN_URL } from '../../common/constants'
import {  
    ICON_COLORS, ICON_NAMES
} from './constants';

/* Component imports */
import NIcon from './../NIcon/NIcon.vue';

export default {
    name: 'nitrozen-alert',
    components: {
        'nitrozen-icon': NIcon
    },
    props: {
        alertWidth: {
            type: String,
            default: ''
        },
        buttonText: {
            type: String,
            default: ''
        },
        buttonType: {
            type: String,
            default: 'default',
            validator: function(value) {
                return ['button', 'default', 'link'].includes(value);
            }
        },
        classNames: {
            type: String,
            default: ''
        },
        displayButton: {
            type: Boolean,
            default: false
        },
        extendedAlert: {
            type: Boolean,
            default: false
        },
        fullWidth: {
            type: Boolean,
            default: true
        },
        href: {
            type: String,
            default: ''
        },
        labelText: {
            type: String,
            default: ''
        },
        linkText: {
            type: String,
            default: ''
        },
        loader: {
            type: Boolean,
            default: false
        },
        state: {
            type: String,
            default: STATES.INFO,
            validator: function(value) {
                return [STATES.ERROR, STATES.INFO, STATES.SUCCESS, STATES.WARN].includes(value);
            }
        }
    },
    computed: {
        classes() {
            return this.setClasses(this.state);
        },
        componentWidth() {
            if(this.alertWidth !== undefined) {
                return this.alertWidth;
            } else if(!this.alertWidth && this.fullWidth) {
                return '100%';
            } else if(!this.alertWidth && !this.fullWidth) {
                return 'fit-content';
            } else {
                return 'fit-content';
            }
        },
        alertIcon() {
            return this.setIcon(this.state);
        },
        iconColor() {
            return ICON_COLORS[this.state];
        },
        isAlertXS() {
            if(
                this.alertWidth && 
                (
                    this.alertWidth < '25%' || 
                    this.alertWidth < '200px' ||
                    this.alertWidth < '20rem'
                )
            ) return true;
            else return false;
        },
        getLoader(){
            return LOADER_CDN_URL
        }
    },
    methods: {
        /**
         * Method to handle click of the button.
         * The function will simply emit the native event with the
         * key name 'click'.
         * 
         * @author Rushabh Mulraj Shah
         * @since 0.0.42
         * @param {Object} event The native event fired by the browser.
         */
        onButtonClick(event) {
            this.$emit('click', event);
        },

        /**
         * Method to set the classnames based on the state provided in the props
         * 
         * @author Rushabh Mulraj Shah
         * @since 0.0.42
         * @param {string} state The state property received as a prop
         */
        setClasses(state) {        
            switch(state) {
                case 'error':
                    return {
                        button: 'n-alert-button-error',
                        buttonLabel: 'n-alert-button-link-error',
                        container: 'n-alert-error',
                        icon: 'n-alert-icon-error'
                    }
                case 'info':
                    return {
                        button: '',
                        buttonLabel: 'n-alert-button-link-info',
                        container: 'n-alert-info',
                        icon: 'n-alert-icon-info'
                    }
                case 'success':
                    return {
                        button: 'n-alert-button-success',
                        buttonLabel: 'n-alert-button-link-success',
                        container: 'n-alert-success',
                        icon: 'n-alert-icon-success'
                    }
                case 'warn':
                    return {
                        button: 'n-alert-button-warn',
                        buttonLabel: 'n-alert-button-link-warn',
                        container: 'n-alert-warn',
                        icon: 'n-alert-icon-warn'
                    }
                default:
                    return {
                        button: '',
                        buttonLabel: 'n-alert-button-link-info',
                        container: 'n-alert-info',
                        icon: 'n-alert-icon-info'
                    }
            }
        },

        /**
         * Method to set the icons to be displayed in an alert based on the state provided in the props
         * 
         * @author Rushabh Mulraj Shah
         * @since 0.0.42
         * @param {string} state The state property received as a prop
         */
        setIcon(state) {
            if(state !== undefined) {
                let selectedState = ICON_NAMES[state];
                return selectedState !== undefined ? selectedState : ICON_NAMES[STATES.INFO]; 
            } else {
                return ICON_NAMES[STATES.INFO];
            }
        }
    }
}
</script>

<style lang="less" scoped>
@import "./../../base/base.less";

.n-alert {
    box-sizing: border-box;
    display: flex;
    align-items: center;
    padding: 1rem;
    gap: 0.8rem;
    min-height: 4.8rem;
    cursor: default;
    border-radius: 1.6rem;
    border: 0.1rem solid @LabelColor;
    color: @TypographyPrimaryColor;
    font-size: @BaseFontSize + 2;
    font-family: @PrimaryFont;

    &:focus {
        outline: none;
    }

    &.n-alert-full-width {
        width: 100%;

        .n-alert-label-text,
        .n-alert-text {
            font-size: @BaseFontSize + 6;
        }
    }

    &.n-alert-link-button-container {
        .n-alert-label-text,
        .n-alert-text {
            font-size: @BaseFontSize + 4;
        }
    }

    &.n-alert-info {
        background: @ColorPrimary20;
        border: 0.1rem solid @ColorPrimary50;

        .n-alert-button {
            background: @ColorPrimary50;
        }
    }

    &.n-alert-success {
        background: @Bubbles;
        border: 0.1rem solid @SuccessColor;
    }

    &.n-alert-warn {
        background: @ColorFeedbackWarning20;
        border: 0.1rem solid @ColorFeedbackWarning50;
    }

    &.n-alert-error {
        background: @ColorFeedbackError20;
        border: 0.1rem solid @ColorFeedbackError50;
    }

    &.n-alert-loader-container {
        background: @ColorPrimary20 !important;
        border: 0.1rem solid @ColorPrimary50;
    }

    &.n-alert-custom-width {
        .n-alert-content {
            column-gap: 0.2rem;
        }
    }

    &.n-alert-xs {
        .n-alert-content {
            flex-direction: column !important;
            row-gap: 0.8rem;
        }
    }

    .n-alert-content {
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        column-gap: 8rem;
        width: 100%;
        align-items: center;

        &.n-alert-extended {
            flex-direction: column;
            align-items: unset;
            justify-content: center;
            row-gap: 1.2rem;

            .n-alert-button-container {
                display: flex;
                justify-content: flex-end;
            }
        }
    }

    .n-alert-icon-text-wrapper {
        display: flex;
        align-items: center;
        column-gap: 0.4rem;

        .n-alert-icon {
            height: 26.6px !important;
            width: 26.6px !important;
        }
    }

    .n-alert-loader {
        height: 2.4rem;
    }

    .n-alert-button-container {
        .n-alert-button {
            display: flex;
            align-items: center;
            justify-content: center;
            border-radius: 25rem;
            padding: 0.4rem 1.2rem;
            color: @WhiteColor;
            font-weight: 700;
            border: none;
            cursor: pointer;

            &:disabled {
                cursor: not-allowed;
                pointer-events: none;
            }

            &.n-alert-button-success {
                background-color: @SuccessColor;
            }

            &.n-alert-button-warn {
                background-color: @ColorFeedbackWarning50;
            }

            &.n-alert-button-error {
                background-color: @ColorFeedbackError50;
            }
        }

        .n-alert-link {
            cursor: pointer;
            font-weight: 700;
            text-decoration: none;

            &.n-alert-button-link-info {
                color: @ColorPrimary50;
            }

            &.n-alert-button-link-success {
                color: @SeaGreen;
            }

            &.n-alert-button-link-warn {
                color: @WarningColor;
            }

            &.n-alert-button-link-error {
                color: @ErrorColor;
            }
        }
    }
}
</style>
