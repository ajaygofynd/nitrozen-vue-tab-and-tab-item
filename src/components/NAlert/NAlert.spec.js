/* Package imports */
import { mount } from "@vue/test-utils";

/* Component imports */
import NAlert from './NAlert.vue';
import NIcon from "../NIcon";

describe('Alert Component', () => {
    it('should render an alert component with a given state', () => {
        const wrapper = mount(NAlert, {
            propsData: {
                state: 'info'
            }
        });
        expect(wrapper.element).toMatchSnapshot();
    });

    it('should render an element that receives correct props', () => {
        const wrapper = mount(NAlert, {
            propsData: {
                state: 'info'
            }
        });
        expect(wrapper.props('state')).toBe('info');
    });

    it('should display success badge with an icon', () => {
        const wrapper = mount(NAlert, {
            propsData: {
                state: 'success'
            }
        })
        expect(wrapper.findComponent(NIcon).exists()).toBeTruthy();
    })

    it('should render the correct element based on the props received', () => {
        const wrapper = mount(NAlert, {
            propsData: {
                state: 'info'
            }
        });
        let component = wrapper.find('.n-alert');
        expect(component.classes()).toContain('n-alert-info');
    });

    it('should render an alert component with the given width if the user has provided a value using alertWidth', () => {
        const wrapper = mount(NAlert, {
            propsData: {
                alertWidth: '35%',
                state: 'success'
            }
        });
        let component = wrapper.find('.n-alert');
        expect(component.element.style.width).toBe('35%');
    });

    it('should make the component into an extended alert if the user has provided the flag', () => {
        let extendedAlert = true;
        const wrapper = mount(NAlert, {
            propsData: {
                extendedAlert,
                state: 'success'
            }
        });
        let component = wrapper.find('.n-alert-content');
        expect(component.classes()).toContain('n-alert-extended');
    });

    it('should have a full width if the user has set the fullWidth flag to true', () => {
        let fullWidth = true;
        const wrapper = mount(NAlert, {
            propsData: {
                fullWidth,
                state: 'success'
            }
        });
        let component = wrapper.find('.n-alert');
        expect(component.classes()).toContain('n-alert-full-width');
    });

    it('should show a button if the user has enabled the displayButton flag', () => {
        let displayButton = true;
        const wrapper = mount(NAlert, {
            propsData: {
                displayButton,
                state: 'warn'
            }
        });
        let component = wrapper.find('.n-alert-button-container');
        expect(component.exists()).toBe(true);
    });

    it('should show a link button if the user has specified "link" as the type of the button', () => {
        let displayButton = true;
        const wrapper = mount(NAlert, {
            propsData: {
                buttonType: 'link',
                displayButton,
                state: 'warn'
            }
        });
        let component = wrapper.find('.n-alert-link');
        expect(component.exists()).toBe(true);
    });

    it('should not show a button if the user has passed true to the loader props', () => {
        let displayButton = true;
        let loader = true;
        const wrapper = mount(NAlert, {
            propsData: {
                buttonType: 'link',
                displayButton,
                loader,
                state: 'warn'
            }
        });
        let component = wrapper.find('.n-alert-button-container');
        expect(component.exists()).toBe(false);
    });

    it('should show a loader if the user has passed true to the loader props', () => {
        let displayButton = true;
        let loader = true;
        const wrapper = mount(NAlert, {
            propsData: {
                buttonType: 'link',
                displayButton,
                loader,
                state: 'warn'
            }
        });
        let component = wrapper.find('.n-alert-loader');
        expect(component.exists()).toBe(true);
    });
})
