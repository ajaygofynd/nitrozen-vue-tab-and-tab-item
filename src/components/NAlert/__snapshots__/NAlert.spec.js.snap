// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Component should render an alert component with a given state 1`] = `
<transition-stub
  name="nitrozen-alert"
>
  <div
    class="n-alert n-alert-info n-alert-full-width"
  >
    <div
      class="n-alert-content"
    >
      <div
        class="n-alert-icon-text-wrapper"
      >
        <div
          class="n-alert-icon"
        >
          <svg
            fill="none"
            height="26.67"
            preserveAspectRatio="none"
            viewBox="0 0 24 24"
            width="26.67"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2C10.0222 2 8.08879 2.58649 6.4443 3.6853C4.79981 4.78412 3.51809 6.3459 2.76121 8.17317C2.00433 10.0004 1.8063 12.0111 2.19215 13.9509C2.578 15.8907 3.53041 17.6725 4.92894 19.0711C6.32746 20.4696 8.10929 21.422 10.0491 21.8079C11.9889 22.1937 13.9996 21.9957 15.8268 21.2388C17.6541 20.4819 19.2159 19.2002 20.3147 17.5557C21.4135 15.9112 22 13.9778 22 12C22 10.6868 21.7413 9.38642 21.2388 8.17317C20.7363 6.95991 19.9997 5.85752 19.0711 4.92893C18.1425 4.00035 17.0401 3.26375 15.8268 2.7612C14.6136 2.25866 13.3132 2 12 2ZM12 5.5C12.2967 5.5 12.5867 5.58797 12.8334 5.7528C13.08 5.91762 13.2723 6.15189 13.3858 6.42597C13.4994 6.70006 13.5291 7.00166 13.4712 7.29264C13.4133 7.58361 13.2704 7.85088 13.0607 8.06066C12.8509 8.27044 12.5836 8.4133 12.2926 8.47118C12.0017 8.52906 11.7001 8.49935 11.426 8.38582C11.1519 8.27229 10.9176 8.08003 10.7528 7.83336C10.588 7.58668 10.5 7.29667 10.5 7C10.5 6.60218 10.658 6.22064 10.9393 5.93934C11.2206 5.65804 11.6022 5.5 12 5.5ZM14 17.5H10C9.73479 17.5 9.48043 17.3946 9.2929 17.2071C9.10536 17.0196 9 16.7652 9 16.5C9 16.2348 9.10536 15.9804 9.2929 15.7929C9.48043 15.6054 9.73479 15.5 10 15.5H11V12.5H10C9.73479 12.5 9.48043 12.3946 9.2929 12.2071C9.10536 12.0196 9 11.7652 9 11.5C9 11.2348 9.10536 10.9804 9.2929 10.7929C9.48043 10.6054 9.73479 10.5 10 10.5H12C12.2652 10.5 12.5196 10.6054 12.7071 10.7929C12.8946 10.9804 13 11.2348 13 11.5V15.5H14C14.2652 15.5 14.5196 15.6054 14.7071 15.7929C14.8946 15.9804 15 16.2348 15 16.5C15 16.7652 14.8946 17.0196 14.7071 17.2071C14.5196 17.3946 14.2652 17.5 14 17.5Z"
              fill="#3535F3"
              id="info"
            />
          </svg>
        </div>
         
        <!---->
         
      </div>
       
      <!---->
    </div>
  </div>
</transition-stub>
`;
